mono/utils/mono-merp.c

```diff
@@ -64,6 +64,61 @@ os_version_string (void)
 	return "";
 }
 
+static gboolean
+starts_with (const char *pre, const char *str)
+{
+    return strncmp (pre, str, strlen (pre)) == 0;
+}
+
+static const char *
+macos_version_string (void)
+{
+	/* Can't allocate in here, could be called from a signal handler in a
+	 * crashed process.
+	 */
+	static const char *version_string;
+	static char buf[256];
+	static const size_t buf_size = sizeof (buf);
+
+	if (version_string)
+		return version_string;
+
+	/* macOS 10.13.6 or later */
+	if (!version_string) {
+		size_t size = 0;
+		if (sysctlbyname ("kern.osproductversion", NULL, &size, NULL, 0) < 0 || size > buf_size) {
```

Comment:
I realize the buffer being filled is an unlikely scenario, but unless you're positive that `size` will include the null terminator i would use `size >= buf_size` to be safe.